code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -258,7 +259,10 @@ var/const/NO_EMAG_ACT = -50
 		id_card.sex = capitalize(G.formal_term)
 	else
 		id_card.sex = "Unset"
-	id_card.set_id_photo(src)
+
+	// Sometimes you can't get full person photo.
+	// Lets delay it to be sure what appearance is ready.
+	addtimer(CALLBACK(id_card, /obj/item/card/id/proc/set_id_photo, src), 2 SECOND)
```

Comment:
That's definitely not an acceptable fix. Figure out when it needs to be called and call it there.